apps/vdjbase-singularity/0.1/cedar/vdjbase.json

```diff
@@ -0,0 +1,116 @@
+{
+  "name": "irplus-vdjbase-cedar-ireceptorgw",
+  "version": "0.1",
+  "label": "vdjbase",
+  "defaultQueue": "default",
+  "defaultNodeCount": 1,
+  "defaultProcessorsPerNode": 1,
+  "defaultMemoryPerNode": "8GB",
+  "defaultRequestedTime": "1:00:00",
+  "shortDescription": "VDJBase (singularity) app on cedar.computecanada.ca",
+  "longDescription": "iReceptor+ VDJBase app using singularity image on Compute Canada's cedar ARC platform",
+  "executionType": "HPC",
+  "executionSystem": "cedar.computecanada.ca-compute-ireceptorgw",
+  "parallelism": "PARALLEL",
+  "checkpointable": false,
+  "deploymentPath": "/tapis/apps/vdjbase-singularity/0.1/cedar/",
+  "deploymentSystem": "cedar.computecanada.ca-storage-ireceptorgw",
+  "templatePath": "vdjbase.sh",
+  "testPath": "test/test.sh",
+  "ontology": [
+    "http://sswapmeet.sswap.info/agave/apps/Application"
+  ],
+  "modules": [
+    "purge"
+  ],
+  "inputs": [
+    {
+      "id": "singularity_image",
+      "details": {
+        "label": "",
+        "description": "Singularity image file",
+        "showAttribute": false
+      },
+      "semantics": {
+        "minCardinality": 1,
+        "maxCardinality": 1,
+        "ontology": [
+          "http://sswapmeet.sswap.info/mime/application/Json"
+        ],
+        "fileTypes": [
+          "text-0"
+        ]
+      },
+      "value": {
+        "default": "",
+        "visible": true,
+        "required": true
+      }
+    },
+    {
+      "id": "rearrangement_file",
+      "details": {
+        "label": "",
+        "description": "Rearrangement AIRR TSV file",
+        "showAttribute": false
+      },
+      "semantics": {
+        "minCardinality": 1,
+        "maxCardinality": 1,
+        "ontology": [
+          "http://sswapmeet.sswap.info/mime/application/Json"
+        ],
+        "fileTypes": [
+          "text-0"
+        ]
+      },
+      "value": {
+        "default": "",
+        "visible": true,
+        "required": true
+      }
+    }
+  ],
+  "parameters": [
+    {
+      "id": "creator",
+      "value": {
+        "visible": true,
+        "required": true,
+        "type": "string",
+        "default": ""
+      },
+      "details": {
+        "label": "Creator",
+        "description": "Username that created this job."
+      },
+      "semantics": {
+        "minCardinality": 1,
+        "maxCardinality": 1,
+        "ontology": [
+          "xsd:string"
+        ]
+      }
+    },
+    {
+      "id": "sample_name",
```

Comment:
is this is a custom field? does it map to an existing AIRR field?

Reply:
Not 100% sure. Pretty sure it is not sample_id, but we don't have enough experience with VDJBase to know for sure how broad an input AIRR TSV file can be. When inferring alleles I am not sure repertoire_id would be sufficient, because I assume it doesn't do TCR and IGH at the same time??? 

Currently it is a required parameter by VDJBase (or at least a helpful one) that is supplied by the user to group the processed output files.

Reply:
I am guessing this would need the [repertoire_id, sample_processing_id, data_processing_id]